scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -0,0 +1,154 @@
+ * - relative_path_to_file: the relative path to the file to analyse coverage
+ *
+ * Example:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt
+ * Example with custom process timeout:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt processTimeout=10
+ *
+ */
+fun main(vararg args: String) {
+  val repoRoot = args[0]
+  val filePath = args[1]
+
+  ScriptBackgroundCoroutineDispatcher().use { scriptBgDispatcher ->
+    val processTimeout: Long = args.find { it.startsWith("processTimeout=") }
+      ?.substringAfter("=")
+      ?.toLongOrNull() ?: 5
+
+    val commandExecutor: CommandExecutor = CommandExecutorImpl(
+      scriptBgDispatcher, processTimeout = processTimeout, processTimeoutUnit = TimeUnit.MINUTES
+    )
+
+    RunCoverage(repoRoot, filePath, commandExecutor, scriptBgDispatcher).execute()
+  }
+}
+
+/**
+ * Class responsible for executing coverage on a given file.
+ *
+ * @param repoRoot the root directory of the repository
+ * @param filePath the relative path to the file to analyse coverage
+ * @param commandExecutor Executes the specified command in the specified working directory
+ * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+ */
+class RunCoverage(
+  private val repoRoot: String,
+  private val filePath: String,
+  private val commandExecutor: CommandExecutor,
+  private val scriptBgDispatcher: ScriptBackgroundCoroutineDispatcher
+) {
+  private val bazelClient by lazy { BazelClient(File(repoRoot), commandExecutor) }
+
+  private val rootDirectory = File(repoRoot).absoluteFile
+  private val testFileExemptionTextProto = "scripts/assets/test_file_exemptions"
+
+  /**
+   * Executes coverage analysis for the specified file.
+   *
+   * Loads test file exemptions and checks if the specified file is exempted. If exempted,
+   * prints a message indicating no coverage analysis is performed. Otherwise, initializes
+   * a Bazel client, finds potential test file paths, retrieves Bazel targets, and initiates
+   * coverage analysis for each test target found.
+   */
+  fun execute(): MutableList<List<String>> {
+    var coverageDataList = mutableListOf<List<String>>()
+    val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptionTextProto)
+      .getExemptedFilePathList()
+
+    val isExempted = testFileExemptionList.contains(filePath)
+    if (isExempted) {
+      println("This file is exempted from having a test file. Hence No coverage!")
+      return mutableListOf()
+    }
+
+    val testFilePaths = findTestFile(repoRoot, filePath)
+    val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
+
+    for (testTarget in testTargets) {
+      val coverageData = RunCoverageForTestTarget(
+        rootDirectory,
+        testTarget.substringBeforeLast(".kt"),
```

Comment:
```suggestion
        testTarget.removeSuffix(".kt"),
```

This is probably a bit clearer.

Reply:
Got it! Done.